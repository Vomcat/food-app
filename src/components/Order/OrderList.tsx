import { useState, useEffect } from "react";

import { OrderItemProps } from "interfaces/Order";

import OrderItem from "./OrederItem";

const OrederList = () => {
  const [orders, setOrders] = useState<OrderItemProps[]>([]);

  const fetchData = async () => {
    try {
      const response = await fetch(" http://localhost:3000/order");
      const data = await response.json();
      setOrders(data);
    } catch (err) {
      throw new Error("Something went wrong");
    }
  };

  useEffect(() => {
    fetchData();
  }, []);

  return (
    <>
      {orders.map((order, index) => (
        <OrderItem
          key={index}
          name={order.name}
          lastName={order.lastName}
          email={order.email}
          phoneNumber={order.phoneNumber}
          city={order.city}
          streetAddress={order.streetAddress}
          streetNumber={order.streetNumber}
          postCode={order.postCode}
          date={order.date}
          orderItems={order.orderItems}
          orderTotalPrice={order.orderTotalPrice}
        />
      ))}
    </>
  );
};

export default OrederList;
