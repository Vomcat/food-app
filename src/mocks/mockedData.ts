export const MOCK_DATA = [
  {
    id: 1,
    image: "/assets/images/item1.png",
    name: "Mniam Mniam Food",
    price: 5,
  },
  {
    id: 2,
    image: "/assets/images/item2.png",
    name: "Mniam Mniam Food as adsdasdasd",
    price: 5,
  },
];
