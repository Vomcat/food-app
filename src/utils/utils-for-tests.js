import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
// As a basic setup, import your same slice reducers
import cartSlice from "store/cartSlice";

const renderWithProviders = function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { cartReducers: cartSlice },
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export { renderWithProviders as render };
export * from "@testing-library/react";
