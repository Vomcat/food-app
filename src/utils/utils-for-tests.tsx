import React, { PropsWithChildren } from "react";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import type { PreloadedState, Store } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import { BrowserRouter } from "react-router-dom";

import cartSlice from "store/cartSlice";
import type { RootState } from "store/store";

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions
  extends Omit<RenderOptions, "wrapper" | "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: Store;
}

const renderWithProviders = (
  ui: React.ReactElement,
  {
    preloadedState = { items: [], totalAmount: 0, totalQuantity: 0 },
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { cartReducers: cartSlice },
    }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <BrowserRouter>
        <Provider store={store}>{children}</Provider>
      </BrowserRouter>
    );
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export { renderWithProviders as render };
export * from "@testing-library/react";
export * from '@testing-library/jest-dom';

