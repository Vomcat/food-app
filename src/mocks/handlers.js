import { rest } from "msw";
import { MOCK_DATA_PRODUCTS, MOCK_DATA_ORDERS } from "mocks/mockedData";

export const handlers = [
  rest.get(`${process.env.REACT_APP_API_HOST}/items`, (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(MOCK_DATA_PRODUCTS));
  }),

  rest.get(`${process.env.REACT_APP_API_HOST}/order`, (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(MOCK_DATA_ORDERS));
  }),
];
