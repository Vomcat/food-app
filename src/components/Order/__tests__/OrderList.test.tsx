import {  render, screen } from "utils/utils-for-tests";

import OrderList from "components/Order/OrderList";
import { MOCK_DATA_ORDERS } from "mocks/mockedData";

describe("Product List render test", () => {
    it("Order data renders correctly", async () => {
      const { asFragment } = render(<OrderList />);
  
      await screen.findByText(MOCK_DATA_ORDERS[0].orderTotalPrice);
  
      expect(asFragment()).toMatchSnapshot();
    });
  });
