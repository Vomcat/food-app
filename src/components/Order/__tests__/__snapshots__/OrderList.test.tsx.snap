// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order component Order data renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe kZTUrz"
  >
    <h3>
       13
    </h3>
    <p>
      Mniam Mniam Food x 4: 20$,  Spicy noodle with shrimps x 2: 14$
    </p>
  </div>
</DocumentFragment>
`;

exports[`Product List render test Order data renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe kZTUrz"
  >
    <h3>
       13
    </h3>
    <p>
      Mniam Mniam Food x 4: 20$,  Spicy noodle with shrimps x 2: 14$
    </p>
  </div>
</DocumentFragment>
`;

exports[`Product List render test renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe kZTUrz"
  >
    <h3>
       13
    </h3>
    <p>
      Mniam Mniam Food x 4: 20$,  Spicy noodle with shrimps x 2: 14$
    </p>
  </div>
</DocumentFragment>
`;
